packages/newspringchurchapp/src/ui/HorizontalContentCardConnected/horizontalContentCardComponentMapper.js

```diff
@@ -13,21 +15,47 @@ const horizontalContentCardComponentMapper = ({
   ...props
 }) => {
   // map typename to the the card we want to render.
-  switch (get(props, '__typename')) {
+  const typeName = get(props, '__typename');
+
+  switch (typeName) {
     case 'ContentSeriesContentItem':
       return <HorizontalHighlightCard {...props} />;
     case 'MediaContentItem':
       return <HorizontalHighlightCard title={hyphenatedTitle} {...props} />;
     case 'WeekendContentItem':
-    case 'DevotionalContentItem':
+    case 'DevotionalContentItem': {
+      let labelText;
+      if (typeName === 'DevotionalContentItem') {
+        labelText = `${get(props, 'seriesConnection.itemIndex', '')} of ${get(
+          props,
+          'seriesConnection.itemCount',
+          ''
+        )}`;
+      } else if (typeName === 'WeekendContentItem') {
+        labelText = `Week ${get(props, 'seriesConnection.itemIndex')}`;
+      } else {
+        labelText = '';
+      }
       return (
-        <HorizontalHighlightCard
-          title={hyphenatedTitle}
-          {...props}
-          coverImage={null}
-          labelText={null}
-        />
+        <ThemeConsumer>
+          {(theme) => (
+            <HorizontalHighlightCard
+              title={hyphenatedTitle}
+              {...props}
+              coverImage={null}
+              LabelComponent={
+                <CardLabel
+                  type={
+                    get(theme, 'type') === 'light' ? 'darkOverlay' : undefined
+                  }
+                  title={labelText}
+                />
+              }
+            />
+          )}
+        </ThemeConsumer>
```

Comment:
turn this into its own component and then drop that into each of the switch statements with their own `labelText`

Reply:
This was such a good idea!